docs/edgeql/lexical.rst

```diff
@@ -92,17 +92,112 @@ They can be used in most places where a short name can appear (such as
 
 
 .. _ref_eql_lexical_const:
 
 Constants
 ---------
 
-.. TODO
+A number of scalar types have literal constant expressions.
+
+
+Strings
+^^^^^^^
+
+Production rules for :eql:type:`str` literal:
+
+.. productionlist:: edgeql
+    string: <str> | <raw_str>
+    str: "'" `str_content`* "'" | '"' `str_content`* '"'
+    raw_str: "r'" `raw_content`* "'" |
+           : 'r"' `raw_content`* '"' |
+           : `dollar_quote` `raw_content`* `dollar_quote`
+    raw_content: <any character different from delimiting quote>
+    dollar_quote: "$" `q_char`* "$"
+    q_char: "A"..."Z" | "a"..."z" | "_" | "0"..."9" |
+          : <high ASCII characters>
+    str_content: <newline> | `unicode` | `str_escapes`
+    unicode: <any printable unicode character not preceded by "\">
+    str_escapes: <see below for details>
+
+The inclusion of "high ASCII" character in the :token:`q_char` in
+practice reflects the ability to use some of the letters with
+diacritics like ``ò`` or ``ü`` in the dollar-quote delimiter.
+
+Here's a list of valid :token:`str_escapes`:
+
+.. _ref_eql_lexical_str_escapes:
+
++--------------------+---------------------------------------------+
+| Escape Sequence    | Meaning                                     |
++====================+=============================================+
+| ``\newline``       | Backslash and newline ignored               |
++--------------------+---------------------------------------------+
+| ``\\``             | Backslash (\\)                              |
++--------------------+---------------------------------------------+
+| ``\'``             | Single quote (')                            |
++--------------------+---------------------------------------------+
+| ``\"``             | Double quote (")                            |
++--------------------+---------------------------------------------+
+| ``\b``             | ASCII backspace (``\x08``)                  |
++--------------------+---------------------------------------------+
+| ``\f``             | ASCII form feed (``\x0C``)                  |
++--------------------+---------------------------------------------+
+| ``\n``             | ASCII newline (``\x0A``)                    |
++--------------------+---------------------------------------------+
+| ``\r``             | ASCII carriage return (``\x0D``)            |
++--------------------+---------------------------------------------+
+| ``\t``             | ASCII tabulation (``\x09``)                 |
++--------------------+---------------------------------------------+
+| ``\xhh``           | Character with hex value hh                 |
++--------------------+---------------------------------------------+
+| ``\uhhhh``         | Character with 16-bit hex value hhhh        |
++--------------------+---------------------------------------------+
+| ``\Uhhhhhhhh``     | Character with 32-bit hex value hhhhhhhh    |
++--------------------+---------------------------------------------+
+
```

Comment:
Add a paragraph about raw strings.  Cross-link that paragraph from edgeql/expressions

Reply:
I'd also add a few examples of valid literals.

* trailing `\`
* multiline strings
* `$$` & `$foo$`
* raw strings

Reply:
Added examples.

There's already a link from the expressions section to the constants section right after the table.